server/app/jobs/service_balancer_job.rb

```diff
@@ -118,6 +118,7 @@ def pending_deploys?(service)
   def lagging_behind?(service)
     return true if service.deployed_at && service.updated_at > service.deployed_at
     return true if service.deployed_at && service.deploy_requested_at && service.deploy_requested_at > service.deployed_at
+    return true if service.deployed_at && service.containers.where(:deploy_rev.ne => service.deployed_at).count > 0
```

Comment:
Ok, have to agree that something is off here... will investigate / add better tests.